pkg/tfbridge/provider.go

```diff
@@ -780,7 +780,9 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	rescfg := MakeTerraformConfigFromInputs(ctx, p.tf, inputs)
 	warns, errs := p.tf.ValidateResource(ctx, tfname, rescfg)
 	for _, warn := range warns {
-		GetLogger(ctx).Warn(fmt.Sprintf("%v verification warning: %v", urn, warn))
+		if err = p.host.Log(ctx, diag.Warning, urn, fmt.Sprintf("%v verification warning: %v", urn, warn)); err != nil {
```

Comment:
This is another weird change that needs its own PR. I missed the in review. The `return err` from a logging interface  feels a bit dangerous. We shouldn't be failing the provider when for some reason the gRPC machinery for sending warnings logs has malfunctioned.  Perhaps this should be reflected in the the Log() methods not returning `error`.